game-core/src/main/java/games/strategy/triplea/ui/menubar/DebugMenu.java

```diff
@@ -18,10 +18,10 @@
     setMnemonic(KeyEvent.VK_D);
 
     final Set<Player> players = frame.getLocalPlayers().getLocalPlayers();
-    final boolean areThereProAIs = players.stream().anyMatch(ProAi.class::isInstance);
+    final boolean areThereProAIs = players.stream().anyMatch(AbstractProAi.class::isInstance);
     if (areThereProAIs) {
-      ProAi.initialize(frame);
-      add(SwingAction.of("Show Hard AI Logs", ProAi::showSettingsWindow))
+      AbstractProAi.initialize(frame);
+      add(SwingAction.of("Show Hard AI Logs", AbstractProAi::showSettingsWindow))
```

Comment:
Interesting observation.
"Fast AI" is a subclass of the new AbstractProAi, previously just ProAi.
So it would fall under the same category. Is this intended?

Reply:
I would suspect so. It would kinda seem Fast AI is not really a true AI but a variant of ProAI. Given that, I would personally look to see if we can inject behavior to a ProAI class.

EG:
```
FastAI extends ProAi{
   FastAI() {
      super(ProAiConfig.builder()
             .battleCalcImplementation(new FastAiBattleCalc())
             .build());
   }
}

LaboriousAi extends ProAi{
   LaboriousAiI() {
      super(ProAiConfig.builder()
             .battleCalcImplementation(new LaboriousBattleCalc())
             .build());
   }
}
```

Or even better, do the above as constructor args.